### PR TITLE
Add python-parameterized

### PR DESCRIPTION
**Pull request by agdespopoulos**

### CODE REVIEW

**Review by tfoote — CHANGES_REQUESTED** [review]
This looks to be available in debian and ubuntu:

https://packages.debian.org/buster/python-parameterized
https://packages.ubuntu.com/bionic/python/python-parameterized

We should use those. Can you link to the other projects for example I'd expect fedora to use `python-parameterized` too. I found a reference here: https://src.fedoraproject.org/rpms/python-parameterized/blob/master/f/python-parameterized.spec

**Comment by tfoote** [comment]
PS when you submit new rules please state your use case and link to the package listings to make checking them easier. https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md

**Comment by agdespopoulos** [comment]
Clearly my package searching skills need improvement, sorry about that. I've pushed with updates for using more native packages where possible.

In addition to the Debian and Ubuntu packages you linked, here's Fedora https://apps.fedoraproject.org/packages/python-parameterized

The python package being added is [Parameterized](https://github.com/wolever/parameterized), which makes writing parameterized tests in unittest/nose possible, making it easier to run test cases across a wide set of inputs and expected outputs.

**Review by tfoote — APPROVED** [review]
That looks like a great package to leverage for testing. 

Thanks for the update. I made a couple of tweaks to simplify the rules and make it future compatible using wildcard syntax.

File: rosdep/python.yaml [hunk 1]
```diff
@@ -2172,6 +2172,42 @@ python-pandas:
...
+python-parameterized:
+  debian:
+    '*': [python-parameterized]
+    jessie:
+      pip:
+        packages: [parameterized]
+    stretch:
+      pip:
+        packages: [parameterized]
+  fedora: [python-arameterized]
```
[the remark this answers is not in the document]
Comment (agdespopoulos) [reply rosdep/python.yaml]: Ugh. Good catch, thanks. Fixed.

**Comment by sloretz** [comment]
@tfoote I notice there are also `python3-parameterized` packages in debian/ubuntu/fedora. Are `bionic_python3:` entries required for new rosdep keys?

**Comment by tfoote** [comment]
@sloretz If python3 support is desired a new key should be created with `python3-parameterized` as the key and python3 package names. The python3_trusty was used in an early attempt to test with python3 but it requires you to use an os_override flag to change your distro detection to be used. We decided not to follow that route but left the legacy keys there to not break anyone who might still be using it. New rules shouldn't follow that pattern.

**Comment by agdespopoulos** [comment]
I can add the python3 rules if you like, but I didn't because I don't use python3.

Is there anything else that I should do to get this ready to be merged?

**Review by sloretz — APPROVED** [review]

**Pull request merged**